back/app/policies/event_policy.rb

```diff
@@ -4,13 +4,39 @@ class EventPolicy < ApplicationPolicy
   class Scope < ApplicationPolicy::Scope
     attr_reader :attendee_id
 
-    def initialize(user_context, scope, attendee_id = nil)
```

Comment:
It is considered bad practice by the Rails community to let policies depend on any other context than the user and the record, see e.g. https://github.com/varvet/pundit?tab=readme-ov-file#additional-context.

An exception was made at some point for the attendee_id, but ideally this should not have happened.

Could we move this filtering logic to the controller or a separate service, rather than the policy?

Reply:
Hmm ok. The thing is that everything that relies on the 'unlisted' behavior is currently handled through policies. I was not aware of this convention (although I was indeed wondering why I had to jump through so many hoops to pass parameters to the policy). But putting this in a policy made more sense to me, since usually filtering services are used more for optional parameters (the users themselves deciding which data they want) rather than access restrictions (we deciding what data users should be able to see or not).

I guess the proper solution would then be to refactor all those instances that rely on similar behavior to filtering services as well... just afraid that changing this is going to take a couple of days